### PR TITLE
DAOS-9167 tools,control: add dmg pool query-targets

### PR DESCRIPTION
**Pull request by kccain**
Before this change libdaos API function daos_pool_query_target()
and the daos utility, "pool query-targets" command was provided
to inspect a given pool engine's targets (to see up/down state,
storage capacity/usage, etc.).

With this change, dmg pool query-targets is added to allow an admin
to inspect a given pool engine's targets.

Examples of the essential parts of the dmg command line:
  * dmg pool query-targets mypool --rank=6 --target-idx=3
  * dmg pool query-targets mypool --rank=0 --target-idx=0,1,5,7
  * dmg pool query-targets mypool -j --rank=0 --target-idx=0,1,5,7

A mgmt daos_server gRPC handler is implemented, and a corresponding
engine drpc handler that interacts with the pool service. The control
library adds a PoolQueryTarget method to initiate the gRPC from the
dmg utility side, and convert the protobuf reply to PoolQueryTargetResp
form for printing. And the new dmg command implementation is added.

Signed-off-by: Kenneth Cain <kenneth.c.cain@intel.com>

### CODE REVIEW

File: src/mgmt/srv_drpc.c [hunk 1]
```diff
@@ -1794,6 +1794,123 @@ ds_mgmt_drpc_pool_query(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
...
+	}
+
+	/* Populate the response */
+	/* array of pointers to Mgmt__PoolQueryTargetInfo */
+	D_ALLOC_ARRAY(resp.infos, req->n_targets);
+	if (resp.infos == NULL)
+		D_GOTO(out_infos, rc = -DER_NOMEM);
+	resp.n_infos = req->n_targets;
+
+	/* array of Mgmt__PooolQueryTargetInfo so we don't have to allocate individually */
```
Comment (kjacque) [inline src/mgmt/srv_drpc.c:1852]: spelling - there's an extra 'o' in pool

**Review by kjacque — COMMENTED** [review]
Very minor comments, otherwise LGTM.

**Review by kccain — COMMENTED** [review]
Thanks @kjacque for the review of the draft PR. I've made the suggested changes locally. Adding another reviewer now, then will push a final version once all done with review feedback.

**Review by kjacque — APPROVED** [review]

**Review by mjmac — APPROVED** [review]
LGTM

**Comment by kccain** [comment]
@daos-stack/daos-gatekeeper requesting landing. The Jenkins run shows passing tests except a valgrind issue affecting all master builds and PRs transitioned the status to failed. As an alternative, I'm also happy to rebase this on a master after the other issue is resolved, to get a fully clean test run, if that is the preferred direction.

**Pull request merged**